ModComponent/Patches/FirestartingPatches.cs

```diff
@@ -1,19 +1,18 @@
-﻿extern alias Hinterland;
-using HarmonyLib;
-using Hinterland;
+﻿using HarmonyLib;
+using Il2Cpp;
 using ModComponent.API.Behaviours;
 
 namespace ModComponent.Patches;
 
 [HarmonyPatch(typeof(FireManager), nameof(FireManager.PlayerStartFire))]//Exists
 internal static class FireManager_PlayerStartFire
 {
-	internal static void Postfix(FireStarterItem starter, bool __result)
+	internal static void Postfix(FireStarterItem starter)
 	{
-		if (!__result)
-		{
-			return;
-		}
+		//if (!__result)
+		//{
+		//	return;
+		//}
```

Comment:
Undo. It's important that the item isn't destroyed when firemaking is unsuccessful.

Reply:
`public void PlayerStartFire(Fire fire, FireStarterItem starter, FuelSourceItem tinder, FuelSourceItem fuel, FireStarterItem accelerant, Action<bool> onDoneStartingFire)`

Method no longer returns a bool, it is void... Thoughts ?

Reply:
Is `onDoneStartingFire` a new parameter?

Reply:
Looks like the method has changed quite a bit.

2.02
`public bool PlayerStartFire(Campfire campfire, FireStarterItem starter, FuelSourceItem tinder, FuelSourceItem fuel, FireStarterItem accelerant, float percentChanceSuccess)`

2.08
`public void PlayerStartFire(Fire fire, FireStarterItem starter, FuelSourceItem tinder, FuelSourceItem fuel, FireStarterItem accelerant, Action<bool> onDoneStartingFire)`

The description for `RuinedAfterUse` is `Set the condition to 0% after the fire starting finished (either successful or not).`
So shouldn't it break regardless of fire success ?

In which case the `if (!__result)` chek was invalid to begin with.